### PR TITLE
Imported PlanetWishlistItem component and implementef the remove func…

### PR DESCRIPTION
**Pull request by DennisAgyekum**
…tion

Employed usesate and useEffect to fetch data from NASA

### CODE REVIEW

**Comment by DennisAgyekum** [comment]
Screenshot 2024-10-24 at 14 53 40
Screenshot 2024-10-24 at 14 54 58
Screenshot 2024-10-24 at 14 55 27
Screenshot 2024-10-24 at 14 55 56

File: app/nasa_collaboration/page.js [hunk 1]
```diff
@@ -1,72 +1,92 @@
...
   useEffect(() => {
     const fetchRoverPhotos = async () => {
-      const roverPhotoResponse = await fetch(NASA_URLs.marsRoverPhoto).then(response => response.json());
-      setRoverPhoto(roverPhotoResponse);
+      try {
+        const response = await fetch(NASA_URLs.marsRoverPhoto);
+        const data = await response.json();
+        setRoverPhoto(data);
+      } catch (error) {
+        console.error("Error fetching Mars rover photo:", error);
```
Comment (babak-f) [inline app/nasa_collaboration/page.js:36]: How is this error communicated to the user? I.e., how will the user know something went wrong? :)

File: app/nasa_collaboration/page.js [hunk 2]
```diff
@@ -1,72 +1,92 @@
...
     };
 
-    fetchRoverPhotos();
+    const fetchAstronomyPicOfTheDay = async () => {
+      try {
+        const response = await fetch(NASA_URLs.astronomyPicOfTheDay);
+        const data = await response.json();
+        setDailyImg(data);
+      } catch (error) {
+        console.error("Error fetching astronomy picture of the day:", error);
```
Comment (babak-f) [inline app/nasa_collaboration/page.js:49]: Same as above

File: app/nasa_collaboration/page.module.css [hunk 3]
```diff
@@ -1,12 +1,66 @@
-.app {
-  background-color: black;
+ .app {
   width: 100vw;
   height: 100vh;
 }
+  .astronomyContainer {
```
Comment (babak-f) [inline app/nasa_collaboration/page.module.css]: In CSS, the convention is to use [kebab case](https://developer.mozilla.org/en-US/docs/Glossary/Kebab_case) for class names :)

File: components/destination/AddWishlistItem.js [hunk 4]
```diff
@@ -3,37 +3,51 @@
...
-  // TASK - React 1 week 3
-  // 1. Add a useState for the handling the <input id="customWishlist" type="text" />
-  // 2. Connect the onThumbnailChange to the <select>
+export const AddWishlistItem = ({ onAddWishlistItem }) => {
+  const [customWishlist, setCustomWishlist] = useState('');
+  const [thumbnail, setThumbnail] = useState('/destination/image-europa.png');
+
+  const handleInputChange = (e) => {
+    setCustomWishlist(e.target.value);
+
```
Comment (babak-f) [inline components/destination/AddWishlistItem.js]: Let's get into the habit of formatting the code before committing it. You can use an extension like Prettier for VS Code.

File: components/ui/Footer.js [hunk 5]
```diff
@@ -25,19 +25,19 @@ export const Footer = () => {
         <h3>Pages</h3>
         <ul>
           <li>
-            {" "}
+            {""}
```
Comment (babak-f) [inline components/ui/Footer.js:28]: What are these - are they needed?

File: components/destination/AddWishlistItem.js (hunk 4 above)
Comment (DennisAgyekum) [reply components/destination/AddWishlistItem.js]: ok, sorry forgot to do that.

File: app/nasa_collaboration/page.module.css (hunk 3 above)
Comment (DennisAgyekum) [reply app/nasa_collaboration/page.module.css]: Thanks, I did not know that, I will change it.

File: app/nasa_collaboration/page.js (hunk 1 above)
Comment (DennisAgyekum) [reply app/nasa_collaboration/page.js:36]: The ui displays the error message now, thanks for the feedback.

**Pull request merged**